src/add-recipe/AddRecipe.js

```diff
@@ -0,0 +1,43 @@
+import React, { useState, useEffect } from 'react';
+
+import recipeListEndpointAdapter from '../shared/recipeListEndpointAdapter';
+import RecipeForm from '../shared/RecipeForm';
+import { Link } from 'react-router-dom';
+import routes from '../routes';
+import { H1, P } from '../shared/styled';
+
+const Step = {
+  EDITING: 'EDITING',
+  SAVING: 'SAVING',
+  SAVED: 'SAVED',
+};
+
+const AddRecipe = () => {
+  const [step, setStep] = useState(Step.EDITING);
+  const [recipeToSave, setRecipeToSave] = useState();
+
+  function saveRecipe(recipe) {
+    setStep(Step.SAVING);
+    setRecipeToSave(recipe);
+  }
+
+  useEffect(() => {
+    if (recipeToSave) {
+      (async () => {
+        await recipeListEndpointAdapter.post({ recipe: recipeToSave });
+        setStep(Step.SAVED);
+      })();
+    }
+  }, [recipeToSave]);
```

Comment:
Everywhere that HTTP requests happen after user interaction, I've done it declaratively by having the side-effect happen inside a `useEffect` triggered by a change in state. My understanding is that this is the preferred React way of doing things, but I'm also aware that in many cases it works fine is you skip `useEffect` entirely and just call endpoints etc. imperatively in event handlers - the web is full of examples of this. Does TK have a preferred position on this?